File: boj/python/15651.py
n, m = map(int, input().split())

arr = [0 for _ in range(8)]
visited = [False for _ in range(8)]


def foo(cnt):
    if cnt == m:
        for i in range(m):
            print(arr[i], end=" ")
        print("")
        return

    for i in range(1, n+1):
        arr[cnt] = i
        foo(cnt + 1)


foo(0)

File: boj/cpp/15651.cpp
#include <iostream>
using namespace std;

int n, m;
int arr[8];

void foo(int cnt)
{
  if (cnt == m)
  {
    for (int i = 0; i < m; i++)
      cout << arr[i] << " ";
    cout << '\n';
    return;
  }

  for (int i = 0; i < n; i++)
  {
    arr[cnt] = i + 1;
    foo(cnt + 1);
  }
}

int main(void)
{
  cin >> n >> m;
  foo(0);
  return 0;
}